controller/eventController.js

```diff
@@ -0,0 +1,223 @@
+                    status: StatusEnum['ERROR'],
+                    message: 'Events not found.'
+                })
+            }
+        }).catch(err => {
+            return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+                status: StatusEnum['ERROR'],
+                message: 'Internal server error.'
+            });
+        });
+        //Regulal users can get all of the events where they are invited    
+    } else if (user && !ValidationUtil.isUserAdmin(user.accessType)) {
+        //TODO: Discuss better and faster solutions for this
+        Invite.find({ userID: new ObjectId(user._id) }).then(invites => {
+            if (!invites) {
+                return res.status(HttpStatus.NOT_FOUND).json({
+                    status: StatusEnum['ERROR'],
+                    message: 'No events for user.'
+                });
+            } else {
+                //First we extract the event id's from invites
+                let eventIDs = [];
+                invites.forEach(invite => {
+                    eventIDs.push(new ObjectId(invite.eventID));
+                });
+                //Now we find all the events for the event ids
+                Event.find({ '_id': { $in: eventIDs } }).then(events => {
+                    if (events) {
+                        return res.status(HttpStatus.OK).json(events);
+                    } else {
+                        return res.status(HttpStatus.NOT_FOUND).json({
+                            status: StatusEnum['ERROR'],
+                            message: 'No events for user.'
+                        });
+                    }
+                }).catch(err => {
+                    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+                        status: StatusEnum['ERROR'],
+                        message: 'Internal server error'
+                    });
+                })
+            }
+        });
+    } else { //Case where the user is not authorized
+        return res.status(HttpStatus.UNAUTHORIZED).json({
+            status: StatusEnum['ERROR'],
+            message: 'Unauthorized access.'
+        });
+    }
+
+
+};
+
+const getById = (req, res) => {
+    const eventId = req.params.id;
+    const user = req.decoded;
+
+    if (!eventId || !user) {
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            status: StatusEnum['ERROR'],
+            message: 'Bad request'
+        });
+    }
+
+    Event.findOne({
+        _id: new ObjectId(eventId),
+        companyID: new ObjectId(user.companyID)
+    }).then(event => {
+        if (event) {
+            //If the user is admin then we can send him the event
+            if (ValidationUtil.isUserAdmin(user.accessType)) {
+                return res.status(HttpStatus.OK).json(event);
+            //If the user is regular, we need to check for the invite
+            } else {
+                Invite.findOne({ userID: new ObjectId(user._id), eventID: new ObjectId(event._id) })
```

Comment:
I think one of the `ResponseTypes` is declined. This means if the user declines the event, that response is stored as well. Should the user still be able to view that event even if he declined the invite? If not add a check for what the responseType is (pending or accepted should be allowed).

Reply:
I will add an enum similar to the one we have for User Types so I check the declined value with it.